### PR TITLE
Parallel Solvers update

### PR DESCRIPTION
**Pull request by Adrian-Diaz**
# Description

## Type of change

Please select all relevant options

- [x] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Formatting and/or style fixes
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Test A : Explicit Solver yaml examples
- [x] Test B : Implicit Solver yaml examples

**Test Configuration**:
* OS version: Linux
* Hardware: WSL2 and Darwin scaling partition
* Compiler: GCC

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] The code builds from scratch with my new changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/Parallel-Solvers/Implicit-Lagrange/Topology_Optimization/Heat_Capacity_Potential_Minimize.h [hunk 1]
```diff
@@ -263,7 +265,7 @@ class HeatCapacityPotentialMinimize_TopOpt : public ROL::Objective<real_t> {
 
     if(nodal_density_flag_){
       FEM_->compute_adjoint_gradients(design_densities, objective_gradients);
-      gp->scale(-1);
+      gp->scale(-1/initial_heat_capacity_potential);
```
Comment (jacob-moore22) [inline src/Parallel-Solvers/Implicit-Lagrange/Topology_Optimization/Heat_Capacity_Potential_Minimize.h:268]: Consider renaming gp to be a meaningful variable name.

File: src/Parallel-Solvers/Implicit-Lagrange/Topology_Optimization/Heat_Capacity_Potential_Minimize.h [hunk 2]
```diff
@@ -316,7 +318,7 @@ class HeatCapacityPotentialMinimize_TopOpt : public ROL::Objective<real_t> {
     const_host_vec_array direction_vector = vp->getLocalView<HostSpace> (Tpetra::Access::ReadOnly);
 
     FEM_->compute_adjoint_hessian_vec(design_densities, objective_hessvec, vp);
-    hvp->scale(-1);
+    hvp->scale(-1/initial_heat_capacity_potential);
```
Comment (jacob-moore22) [inline src/Parallel-Solvers/Implicit-Lagrange/Topology_Optimization/Heat_Capacity_Potential_Minimize.h:321]: Same as above.  Consider renaming hpv to carry more meaning.

File: src/Parallel-Solvers/Implicit-Lagrange/Topology_Optimization/Heat_Capacity_Potential_Minimize.h (hunk 1 above)
Comment (Adrian-Diaz) [reply src/Parallel-Solvers/Implicit-Lagrange/Topology_Optimization/Heat_Capacity_Potential_Minimize.h:268]: Those names remain consistent with the ROL arguments they represent.

Comment (jacob-moore22) [reply src/Parallel-Solvers/Implicit-Lagrange/Topology_Optimization/Heat_Capacity_Potential_Minimize.h:268]: In that case, what does gp represent?

**Comment by jacob-moore22** [comment]
Do the integrated test for the SGH solver still pass?  If so, please note that under the testing section above.

File: src/Parallel-Solvers/Implicit-Lagrange/Topology_Optimization/Heat_Capacity_Potential_Minimize.h (hunk 1 above)
Comment (Adrian-Diaz) [reply src/Parallel-Solvers/Implicit-Lagrange/Topology_Optimization/Heat_Capacity_Potential_Minimize.h:268]: Pointer to the gradient vector

Comment (jacob-moore22) [reply src/Parallel-Solvers/Implicit-Lagrange/Topology_Optimization/Heat_Capacity_Potential_Minimize.h:268]: Ah.  In that case, consider adding a comment when it is created noting that, or rename it to be grad_ptr so future us doesn't have to puzzle it out.

Comment (Adrian-Diaz) [reply src/Parallel-Solvers/Implicit-Lagrange/Topology_Optimization/Heat_Capacity_Potential_Minimize.h:268]: lets do that in another PR, considering its the same pattern in every single constraint and objective file.

Comment (jacob-moore22) [reply src/Parallel-Solvers/Implicit-Lagrange/Topology_Optimization/Heat_Capacity_Potential_Minimize.h:268]: Sounds good.  I just know that future knuckle draggers (myself among them) will take psychic damage from two letter variable names when reading the code lol.

File: src/Parallel-Solvers/Implicit-Lagrange/Topology_Optimization/Heat_Capacity_Potential_Minimize.h (hunk 2 above)
Comment (jacob-moore22) [reply src/Parallel-Solvers/Implicit-Lagrange/Topology_Optimization/Heat_Capacity_Potential_Minimize.h:321]: Just for my sanity, what does hvp mean?  My guess is hessian vector pointer?

**Review by jacob-moore22 — APPROVED** [review]

**Pull request merged**